### PR TITLE
feat(umaverse): Connect wallet

### PR DESCRIPTION
**Pull request by Gamaranto**
**Motivation**
This PR adds the useConnection hook to umaverse, it does not touch the logic for injecting the test provider as I wasn't unsure how that is used exactly. 

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: utils/constants.ts [hunk 1]
```diff
@@ -41,3 +43,75 @@ export const CATEGORIES_PLACEHOLDERS: Record<Category, string> = {
 };
 
 export const KNOWN_LSP_ADDRESS = "0x372802d8A2D69bB43872a1AABe2bd403a0FafA1F";
+
+export const infuraId = process.env.NEXT_PUBLIC_INFURA_ID || "";
```
Comment (daywiss) [inline utils/constants.ts:47]: not sure if we have a place to document this, but should always try to document new envs

File: utils/constants.ts [hunk 2]
```diff
@@ -41,3 +43,75 @@ export const CATEGORIES_PLACEHOLDERS: Record<Category, string> = {
...
+  }
+};
+export function onboardBaseConfig(_chainId?: number): Initialization {
+  const chainId = _chainId ?? 1;
+  const infuraRpc = `https://${getNetworkName(
+    chainId
+  )}.infura.io/v3/${infuraId}`;
+
+  return {
+    dappId: process.env.NEXT_PUBLIC_ONBOARD_API_KEY || "",
```
Comment (daywiss) [inline utils/constants.ts:71]: doc this

File: utils/constants.ts [hunk 3]
```diff
@@ -41,3 +43,75 @@ export const CATEGORIES_PLACEHOLDERS: Record<Category, string> = {
...
+          rpcUrl:
+            chainId === 1
+              ? "https://mainnet-eth.token.im"
+              : "https://eth-testnet.tokenlon.im",
+          preferred: true,
+        },
+        { walletName: "coinbase", preferred: true },
+        {
+          walletName: "portis",
+          apiKey: process.env.NEXT_PUBLIC_PORTIS_API_KEY,
```
Comment (daywiss) [inline utils/constants.ts:88]: doc this

**Review by daywiss — COMMENTED** [review]
@Gamaranto this looks like a copy and paste from claim dapp (which is fine), just want to know if there were any significant differences. other than that it looks good.

File: utils/constants.ts (hunk 1 above)
Comment (Tulun) [reply utils/constants.ts:47]: Definitely README. it should probably be noted in the PR even?

File: utils/constants.ts (hunk 2 above)
Comment (Tulun) [reply utils/constants.ts:71]: Same as above.

**Review by Tulun — APPROVED** [review]
Add documentation as David stated.

Given this is just a port over more or less I think we can just add it in.

**Comment by Gamaranto** [comment]
Added a table to document environment variables, I suggest we add new ones there as soon, as they are added. To see how it looks like on Github, see https://github.com/Gamaranto/umaverse/tree/connect-wallet

**Pull request merged**